cstar_ocean/input_dataset.py

```diff
@@ -193,41 +273,61 @@ def check_exists_locally(self, local_dir: str) -> bool:
         return self.exists_locally
 
 
-class ModelGrid(InputDataset):
+class ROMSInputDataset(InputDataset, ABC):
```

Comment:
It's not entirely clear what the point of differentiating between a `ROMSInputDataset` and an `InputDataset` is, as there is no common logic in `ROMSInputDataset`. Are you anticipating there being some logic that is relevant for all input datasets that are needed for ROMS, but not for input datasets needed for non-ROMS components?

Reply:
Yes I anticipate that in future this will be needed. Tools-Roms (the fortran ones) currently operate on any and all things covered by `ROMSInputDataset`, for example. I think if anything that `InputDataset` is the redundant class here